atom/browser/api/lib/app.coffee

```diff
@@ -34,6 +34,27 @@ app.setAppPath = (path) ->
 app.getAppPath = ->
   appPath
 
+app.hide = ->
+  if process.platform is 'darwin'
+    Menu.sendActionToFirstResponder('hide:')
+  else
+    BrowserWindow.getAllWindows().forEach window ->
+      window._wasMinimized = window.isMinimized()
+      window._wasFocused = window.isFocused()
+
+      window.hide()
+
+app.show = (focus = false) ->
+  if process.platform is 'darwin'
+    action = focus ? 'activateIgnoringOtherApps:' : 'unhide:'
```

Comment:
The `activateIgnoringOtherApps` was already implemented in the `Browser::Focus` native code, if you really want to expose it you can map `app.focus` to it.